src/line_agg.rs

```diff
@@ -135,7 +135,11 @@ impl<K, C> Logic<K, C> {
     }
 }
 
-impl<T, K, C> LineAgg<T, K, C> {
+impl<T, K, C> LineAgg<T, K, C>
+where
+    T: Stream<Item = (K, Bytes, C)> + Unpin,
+    K: Hash + Eq + Clone,
+{
```

Comment:
I added this to improve the error messages we get when `LineAgg` was created with something that didn't `impl Stream` properly. This catches it a bit earlier and makes things less confusing.

Reply:
Github doesn't merge the comments :( See https://github.com/timberio/vector/pull/4719#discussion_r511015595
Let's just move the `new` to the block below. This is just a nit though.

Reply:
We can't move `new` because it's not part of the `Stream` trait. Again, this isn't required at all, it just helps give a more useful error message for users of `LineAgg`.